### PR TITLE
telemetry: Add billing categorization to cody events 2

### PR DESCRIPTION
**Pull request by dadlerj**
Adds billing metadata to some events per RFC 978: https://docs.google.com/document/d/1EvyH1kaU-fsn59h-DyoaO2Qy4C2TYLmDGjG73bdb2V4/edit#heading=h.trqab8y0kufp

follow on to https://github.com/sourcegraph/cody/pull/5493 and https://github.com/sourcegraph/sourcegraph/pull/529

## Test plan

CI

### CODE REVIEW

**Comment by dadlerj** [comment]
Hey @pkukielka ! I'm not sure how to diagnose this—when I run `./gradlew check` and `./gradlew integrationTest` locally, they both pass with no errors. I don't see any issues on main either, so it's not like I just need to rebase...

I'd love any recommendations. Thank you. Again, feel free to reassign if there's someone else I should talk to!

File: src/main/java/com/sourcegraph/cody/agent/protocol/TelemetryEventParameters.kt [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
+}
+
+enum class BillingCategory {
+  CORE,
+  BILLABLE
+}
+
+data class BillingMetadata(val product: BillingProduct, val category: BillingCategory)
+
 data class TelemetryEventParameters(
```
Comment (mkondratek) [inline src/main/java/com/sourcegraph/cody/agent/protocol/TelemetryEventParameters.kt:14]: btw, this protocol class is not generated 
```
interface TelemetryEvent {
    feature: string
    action: string
    parameters?:
        | TelemetryEventParameters<{ [key: string]: number }, BillingProduct, BillingCategory>
        | undefined
        | null
}
```
It looks like our generating tool does not support fields like `parameters` in this case. I'd be nice to get it working.

**Review by mkondratek — APPROVED** [review]
hi @dadlerj 👋 sorry for the delay 🙇 

I rebased your branch. I tried your changes. The event are properly sent to the agent 🟢

**Comment by dadlerj** [comment]
Thank you @mkondratek !

**Pull request merged**